src/pages/SudoPrompt.vue

```diff
@@ -0,0 +1,103 @@
+<!--
+  - This is a modal dialog displayed before we ask the user for a sudo password
+  - to explain why we're asking for it.
+  -->
+
+<template>
+  <div class="contents">
+    <h2>Administrative Access Required</h2>
+    <p>
+      Rancher Desktop requires administrative credentials ("sudo access") in
+      order to provide a better experience.  We would like to have access for
+      the following:
+    </p>
+    <ul>
+      <li v-for="item in explanations" :key="item">
+        <tt v-text="item" />
+      </li>
+    </ul>
+    <p>
+      We will display the actual prompt once this window is closed.  Cancelling
+      the password prompt will cause Rancher Desktop to run in reduced
+      functionality mode, but it should still start.
+    </p>
+    <checkbox
+      id="suppress"
+      v-model="suppress"
+      label="Always run without administrative access"
+    />
+    <button ref="accept" class="role-primary" @click="close">
+      OK
+    </button>
+  </div>
+</template>
+
+<script lang="ts">
+import { ipcRenderer } from 'electron';
+import Vue from 'vue';
+import Checkbox from '@/components/form/Checkbox.vue';
+
+export default Vue.extend({
+  components: { Checkbox },
+  layout:     'dialog',
+  data() {
+    return {
+      explanations:    [] as string[],
+      suppress:        false,
+      preferredWidth:  0,
+      preferredHeight: 0,
+    };
+  },
+  mounted() {
+    // We expect some help from the main process to set up the window:
+    // Main Process                Window
+    // (load window)          -->                        Creating the window.
+    //                        <--  "sudo-prompt/load"    The code is loaded.
+    // "sudo-prompt/details"  -->                        Provide explanations.
+    // "sudo-prompt/size"     -->                        Obtaining the window size.
+    //                        <--  "sudo-prompt/ready"   The window is rendered.
+    //                        <--  "sudo-prompt/closed"  The window is closed.
+
+    ipcRenderer.on('sudo-prompt/details', async(event, explanations) => {
+      this.$data.explanations = explanations;
+      await this.$nextTick();
+      this.preferredWidth = Math.max(this.preferredWidth, document.documentElement.scrollWidth);
+      this.preferredHeight = Math.max(this.preferredHeight, document.documentElement.scrollHeight);
+      window.resizeTo(this.preferredWidth, this.preferredHeight);
+      ipcRenderer.send('sudo-prompt/ready');
+    });
+    ipcRenderer.on('sudo-prompt/size', (event, { width, height }) => {
+      this.preferredWidth = Math.max(this.preferredWidth, width);
+      this.preferredHeight = Math.max(this.preferredHeight, height);
+      window.resizeTo(this.preferredWidth, this.preferredHeight);
+    });
+    window.addEventListener('close', () => {
+      ipcRenderer.send('sudo-prompt/closed', this.suppress);
+    });
+    (this.$refs.accept as HTMLButtonElement)?.focus();
```

Comment:
Hmm, even after hacking up `directives.js` to find the correct element, it still fails to focus properly — do you know _when_ the focus happens? If I select the element in the developer console afterwards and call `focus()` on it, it _does_ work, so I think it's timing-related.

Reply:
We call `focus` on the `inserted` hook, which, according to the vue docs is

> called when the bound element has been inserted into its parent node (this only guarantees parent node presence, not necessarily in-document).

We can also play with the `componentUpdated` or `update` hooks.. but I'm fine with sticking to the current implementation that you have here and making this change later. I wouldn't spend too much effort on it now if it's not immediately working.

Reply:
Ah, yeah, you can't focus anything that's not a descendant of the document element (because focus is a UI state, so it only applies once the user _can_ focus it).

We can think about it later.